### PR TITLE
remove min/maxAllowed

### PR DESCRIPTION
**Pull request by AndreasBurger**
**How to categorize this PR?**

/area control-plane
/kind enhancement
/platform azure

**What this PR does / why we need it**:
Remove VPA (min|max)Allowed settings.

minAllowed may cause wasted resources while maxAllowed may cause under-reservation and hence noisy neighbor issues.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```other operator
The provider-azure extension no longer configures min/maxAllowed in any managed VPA resource.
```

### CODE REVIEW

File: charts/gardener-extension-provider-azure/templates/vpa.yaml [hunk 1]
```diff
@@ -5,13 +5,6 @@ metadata:
   name: {{ include "name" . }}-vpa
   namespace: {{ .Release.Namespace }}
 spec:
-  {{- if .Values.vpa.resourcePolicy }}
-  resourcePolicy:
-    containerPolicies:
-    - containerName: '*'
-      minAllowed:
-        memory: {{ required ".Values.vpa.resourcePolicy.minAllowed.memory is required" .Values.vpa.resourcePolicy.minAllowed.memory }}
-  {{- end }}
```
Comment (kon-angelo) [inline charts/gardener-extension-provider-azure/templates/vpa.yaml:14]: same as above

File: charts/gardener-extension-admission-azure/charts/runtime/templates/vpa.yaml [hunk 2]
```diff
@@ -5,13 +5,6 @@ metadata:
   name: {{ include "name" . }}-vpa
   namespace: {{ .Release.Namespace }}
 spec:
-  {{- if .Values.global.vpa.resourcePolicy }}
-  resourcePolicy:
-    containerPolicies:
-    - containerName: '*'
-      minAllowed:
-        memory: {{ required ".Values.global.vpa.resourcePolicy.minAllowed.memory is required" .Values.global.vpa.resourcePolicy.minAllowed.memory }}
-  {{- end }}
```
[the remark this answers is not in the document]
Comment (AndreasBurger) [reply charts/gardener-extension-admission-azure/charts/runtime/templates/vpa.yaml:14]: Just to be sure: that is a change in behaviour

**Review by kon-angelo — APPROVED** [review]
/lgtm

**Pull request merged**